### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by saulorfreitas**
Main changes in this version:

1)	adding namelist options in GF MONAN to activate the cold pool scheme,… 
2)	adding subsidence spreading
3)	including mpi exchange for 1-d array
4)	including subsidence tendency for U and V
5)	changing packages dependence of buoyx and cnvcf scalars
6)	including directory with MONAN specific namelists
7)	overwritten the GF namelist option with MPAS namelist_atmosphere spec… 
8)	fix access to array not allocated when using CAM SW radiation scheme
9)	using the tendency potential temp to limit the max and min heating rates
10)	adding routine for applying turbulent mixing in the dry PBL for extra… 
11)	minor changes

### CODE REVIEW

File: src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+       !  c(*) upper diagonal (Ai,i+1)                                                                  
+       !  d                                                                                               
+       ! Output                                                                                           
+       !  x     results                                                                                   
+       !ccccccccccccccccccccccccccccccc                                                                   
+        integer,intent(in)   :: kte
+        integer, parameter   :: kts=1
+        real(kind_phys), dimension(kte) :: a,b,c,d
+        real(kind_phys), dimension(kte), intent(out) :: x
+        integer :: in
```
Comment (joaomas) [inline src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F:285]: "in" is reserved word

File: src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+      type(mpas_pool_type),intent(in) :: diag_physics
+      type(mpas_pool_type),intent(in) :: state
+      type(mpas_pool_type),intent(in) :: mesh
+      integer,intent(in) :: itimestep,time_lev
+      integer,intent(in) :: its,ite,kts,kte
+      real, intent(in) :: dt_pbl
+      
+      !--- local vars
+      integer, parameter :: max_nscalars = 100
+      integer :: i,j,k,initflag,nscalars,isc
```
Comment (joaomas) [inline src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F:29]: "DTN 4.7 - Variable names made up of more than one word must use the snake_case style, that is, the words in lowercase and the rest separated by underscores."

initflag = init_flag ?

File: src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+           local_scalars(k,isc) = max(1.e-12,x(k))
+        enddo
+     enddo
+
+   end subroutine mix_scalars
+!-----------------------------------------------------------------------------
+   subroutine mix_scalars_opt(kts,kte,delt,dz,rho,dfh,nscalars,local_scalars)
+     integer, intent(in) :: kts,kte,nscalars
+     real(kind_phys), intent(in) :: delt
+     real(kind_phys), dimension(kts:kte), intent(in) :: dfh,dz,rho
```
Comment (joaomas) [inline src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F:209]: "DTN 01 - 4.37 - The preferred mechanism for dynamic memory allocation is automatic arrays.

Example:
real(kind_phys), dimension(kts:kte), intent(in) :: dfh,dz,rho

Suggestion:
real(kind_phys), dimension(:), intent(in) :: dfh,dz,rho

**Review by joaomas — CHANGES_REQUESTED** [review]
Some discrepancies were found in the Code Review.

_"DTN 01 - 4.6 - All programs, modules or procedures must have an implicit none declaration at the beginning"_

**It will require corrections from the developer.
Please check all new subroutines.**

**The Rules below will be ignored at this time and accepted.**

_"DTN 01 - 4.19 - As a general rule for constructing fonts, the indentation was defined with 4 (four) spaces. Do not use tabs for this purpose, as each editor can open the indentation with tabs in the wrong way. Defining an indentation leaves the clearest code."_

_"DTN 01 - 4.37 - The preferred mechanism for dynamic memory allocation is automatic arrays."_

**Please check the comments on the lines within the revised code.**

File: src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+      type(mpas_pool_type),intent(in) :: mesh
+      integer,intent(in) :: itimestep,time_lev
+      integer,intent(in) :: its,ite,kts,kte
+      real, intent(in) :: dt_pbl
+      
+      !--- local vars
+      integer, parameter :: max_nscalars = 100
+      integer :: i,j,k,initflag,nscalars,isc
+      integer, dimension(max_nscalars) :: index_scalar
+      real(kind=kind_phys):: dzk
```
[the remark this answers is not in the document]
Comment (deniseiras) [reply src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F:31]: it's not mandatory right now. Saulo understand the appointment. So, we consider this resolved,

File: src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+   private
+   public :: driver_pbl_scalars
+   contains
+   !
+   !================================================================================================================= 
+   ! A routine to calculate the dry mixing in PBL for scalars not included in the MYNN routine.
+   ! This is strongly based on "mynn_mix_chem" routine in the file 'bl_mynn_subroutines.F'
+   ! Saulo R. Freitas - 19 Sep 2024 (send comments to saulo.freitas@inpe.br)
+   !================================================================================================================= 
+   subroutine driver_pbl_scalars (itimestep,diag_physics,configs,state,mesh,time_lev,its,ite,kts,kte,dt_pbl)
```
[the remark this answers is not in the document]
Comment (deniseiras) [reply src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F:16]: implicit none inserted

File: src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F (hunk 2 above)
Comment (deniseiras) [reply src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F]: not solved

**Comment by saulorfreitas** [comment]
Just delete it and accepts the PR. This namelist is now in the new directory 'namelist_monan'

File: src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+          enddo 
+
+      enddo ! loop i
+
+      if(allocated(local_scalars)) deallocate(local_scalars)
+
+   end subroutine driver_pbl_scalars
+!-----------------------------------------------------------------------------
+
+   subroutine mix_scalars(kts,kte,delt,dz,rho,dfh,nscalars,local_scalars)
```
[the remark this answers is not in the document]
Comment (deniseiras) [reply src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F:142]: not solved (mix_scalars without implicit none)

File: src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+        call tridiag3(kte,a,b,c,d,x)
+        
+        do k=kts,kte
+           local_scalars(k,isc) = max(1.e-12,x(k))
+        enddo
+     enddo
+
+   end subroutine mix_scalars
+!-----------------------------------------------------------------------------
+   subroutine mix_scalars_opt(kts,kte,delt,dz,rho,dfh,nscalars,local_scalars)
```
[the remark this answers is not in the document]
Comment (deniseiras) [reply src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F:208]: not solved (mix_scalars_opt without implicit none)

File: src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F (hunk 3 above)
Comment (deniseiras) [reply src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F]: not solved.

File: src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F [hunk 8]
```diff
@@ -0,0 +1,305 @@
...
+        
+        do k=kts,kte
+           local_scalars(k,isc) = max(1.e-12,x(k))
+        enddo
+     enddo
+
+   end subroutine mix_scalars_opt
+!-----------------------------------------------------------------------------
+!>\ingroup gsd_mynn_edmf
+   subroutine tridiag3(kte,a,b,c,d,x)
```
[the remark this answers is not in the document]
Comment (deniseiras) [reply src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F:272]: not solved

File: src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F (hunk 1 above)
Comment (deniseiras) [reply src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F]: not solved

File: src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F (hunk 4 above)
[the remark this answers is not in the document]
Comment (deniseiras) [reply src/core_atmosphere/physics/physics_monan/module_bl_mixingscalars.F:32]: resolved

**Review by deniseiras — APPROVED** [review]
All problems resolved

**Pull request merged**